tests/e2e/main_test.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Mia srl
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build conformance
+
+//nolint:thelper
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/envfuncs"
+	"sigs.k8s.io/e2e-framework/support/kind"
+)
+
+type namespaceCtxKey string
+
+var (
+	testenv env.Environment
+)
+
+func TestMain(m *testing.M) {
+	testenv = env.New()
+
+	// Specifying a run ID so that multiple runs wouldn't collide.
+	runID := envconf.RandomName("ns", 4)
+
+	kindClusterName := "mlp-e2e-tests"
+	kindImageName := "kindest/node:v1.28.9@sha256:dca54bc6a6079dd34699d53d7d4ffa2e853e46a20cd12d619a09207e35300bd0"
+	if nameFromEnv, found := os.LookupEnv("KIND_NODE_IMAGE"); found {
+		kindImageName = nameFromEnv
+	}
+
+	imageOpts := kind.WithImage(kindImageName)
+
+	kindConfigPath := filepath.Join("testdata", "kind.yaml")
+	crdsPath := filepath.Join("testdata", "crds")
+
+	// Use pre-defined environment funcs to create a kind cluster prior to test run
+	testenv.Setup(
+		envfuncs.CreateClusterWithConfig(kind.NewProvider(), kindClusterName, kindConfigPath, imageOpts),
+		envfuncs.SetupCRDs(crdsPath, "*"),
+	)
+
+	testenv.BeforeEachTest(func(ctx context.Context, cfg *envconf.Config, t *testing.T) (context.Context, error) {
+		return createNSForTest(ctx, t, cfg, runID)
+	})
+
+	testenv.AfterEachTest(func(ctx context.Context, cfg *envconf.Config, t *testing.T) (context.Context, error) {
+		return deleteNSForTest(ctx, t, cfg)
+	})
+
+	// Use pre-defined environment funcs to teardown kind cluster after tests
+	testenv.Finish(
+		// envfuncs.ExportClusterLogs(kindClusterName, "logs"),
+		envfuncs.TeardownCRDs(crdsPath, "*"),
+		// envfuncs.DestroyCluster(kindClusterName),
```

Comment:
Why those are commented?

Reply:
Is faster to enable them when doing troubleshooting if something is not going correctly during testing, but they add noise and bloat if left on all the time